### DISCUSSION TITLE
3 or 3.3V implementation

### DISCUSSION THREAD

**Discussion by pyrophreek** [micronucleus/micronucleus#246] [Q&A]
I am working on a project where I would like to use a tiny84 or Tiny 841 to control an SPI epaper driver, and I would like the system to run at 3v so it can be powered by a coin cell. Will micronucleus work under these conditions? Will the default configuration work or do I need to make changes?

**Comment by ArminJo** [comment]
See https://www.instructables.com/Arduino-Open-Window-Detector-for-Winter/ 
This is an Attiny85 running with a coin cell and programmed with micronucleus.

**Comment by pyrophreek** [comment]
Unfortunately it looks like that instructable destroys the USB capabilities in favor of lower power consumption.  
I would like to keep the USB functionality of the micronucleus bootloader for programmability, while using a 3V regulator when connected to USB.  I then have a switch to swap to battery power of the coin cell.
Unfortunately I can't have the power rail run at 5V as the other chips on the board are not rated for that voltage.

**Reply by cpldcpu** [reply]
Well, you could try. Officially the Attiny85 is not rated for 16MHz at 3V, but it may still work nevertheless. 

Otherwise you could add some diodes to switch to 5V supply when USB is used.

**Reply by pyrophreek** [reply]
Yeah, I figured that was the case. When I tried uploading the default t84 micronucleus binary it didn't work.
Is there a way for micronucleus to work at 8MHz or slower? or is it just unable to keep up with the USB protocol at that point?

**Reply by ArminJo** [reply]
> Unfortunately it looks like that instructable destroys the USB capabilities in favor of lower power consumption.

From where did you deduct this information?

**Reply by pyrophreek** [reply]
"Since we want to save power, the board clock is switched to 1 MHz in our setup() you may choose Digispark (1mhz - No USB) as board in the Tools menu."
also it mentions having to switch to ICSP programmer.

Even then, you would be running it at 5V when you plug it in to program it.

**Reply by cpldcpu** [reply]
> 
> 
> Yeah, I figured that was the case. When I tried uploading the default t84 micronucleus binary it didn't work.
> Is there a way for micronucleus to work at 8MHz or slower? or is it just unable to keep up with the USB protocol at that point?

No, minium is 12 Mhz or better 12.8 MHz
